src/solvers/slepc_eigen_solver.C

```diff
@@ -633,15 +633,29 @@ void SlepcEigenSolver<T>:: set_slepc_position_of_spectrum()
       ierr = EPSSetWhichEigenpairs (_eps, EPS_LARGEST_MAGNITUDE);  LIBMESH_CHKERR(ierr); return;
     case SMALLEST_MAGNITUDE:
       ierr = EPSSetWhichEigenpairs (_eps, EPS_SMALLEST_MAGNITUDE); LIBMESH_CHKERR(ierr); return;
+    case TARGET_MAGNITUDE:
+      {
+      ierr = EPSSetTarget(_eps, this->_target_val); LIBMESH_CHKERR(ierr);
+      ierr = EPSSetWhichEigenpairs (_eps, EPS_TARGET_IMAGINARY); LIBMESH_CHKERR(ierr); return;
```

Comment:
Is this right?  You are in `TARGET_MAGNITUDE` case, but you're passing `EPS_TARGET_IMAGINARY` to SLEPc...